metrics/options.go

```diff
@@ -17,10 +18,20 @@ const (
 type Option func(*options) error
 
 type options struct {
-	errorHandler     errors.ErrorHandler
-	dsdEndpoint      string
-	metricSampleRate float64
-	tags             []string
+	errorHandler errors.ErrorHandler
+	dsdEndpoint  string
+	sampleRate   float64
+	tags         []string
+}
+
+// MetricOptions represents a configuration option for metrics.
+type MetricOptions func(*metricOptions) error
```

Comment:
Should we use singular here for consistency @alfredosa, @dalryan and @bendiknesbo. Similar to https://github.com/coopnorge/go-datadog-lib/blob/62cef1a33df62a3128af2bfa5d8ba4086eae8631/metrics/options.go#L16-L17

```suggestion
// MetricOption represents a configuration option for metrics.
type MetricOption func(*metricOptions) error
```

Reply:
Yes, I'd suggest we use the singular form here

Reply:
yeahhh reads better! :)